### DISCUSSION TITLE
OpenID-Connect and UserDetails

### DISCUSSION THREAD

**Discussion by petrroda** [micronaut-projects/micronaut-security#534] [Q&A]
Hi,

how is the generation of UserDetails working with openid-connect?

The **UserDetails** generated by **DefaultOpenIdUserDetailsMapper** are always overwritten by a call of
**JwtTokenValidator** from the **TokenAuthenticationFetcher**.

Greets
Petr

**Comment by sdelamo** [comment]
@petrroda Not sure what you mean exactly. 

Typically, when you work with OpenID Connect in Micronaut you can either get the details from the `idtoken` and generate a new token (hence  `DefaultOpenIdUserDetailsMapper` mapping the response to a user details which is then used to generate a new token) or you could use the id token directly.
